### PR TITLE
Fix/sgw mercury fixes2

### PR DESCRIPTION
**Pull request by sridevi-15**
#### Fixes #.

- [x] Ran `flake8`
- [x] Ran `run_repo_tests.sh`

#### Changes proposed in this pull request:

- All config changes made for mercury to work with mercury builds

### CODE REVIEW

File: testsuites/syncgateway/functional/tests/test_log_rotation_new.py [hunk 1]
```diff
@@ -680,6 +681,97 @@ def test_log_logLevel_invalid(params_from_base_test_setup, sg_conf_name):
     pytest.fail("SG shouldn't be started!!!!")
 
 
+@pytest.mark.syncgateway
+@pytest.mark.logging
+@pytest.mark.parametrize("sg_conf_name", ["log_rotation_new"])
+def test_rotated_logs_size_limit(params_from_base_test_setup, sg_conf_name):
+    """
+    @summary
+    Test to check rotated log size limit with 100MB( 1024Mb by default)
```
Comment (hrajput89) [inline testsuites/syncgateway/functional/tests/test_log_rotation_new.py]: can we add a more detailed summary?

File: libraries/testkit/user.py [hunk 2]
```diff
@@ -297,7 +299,7 @@ def update_docs(self, num_revs_per_doc=1, retries=False):
         if len(self.cache.keys()) == 0:
             log.warning("Unable to find any docs to update")
 
-        with concurrent.futures.ThreadPoolExecutor(max_workers=settings.MAX_REQUEST_WORKERS) as executor:
+        with concurrent.futures.ThreadPoolExecutor(max_workers=25) as executor:
```
Comment (mgcouchbase) [inline libraries/testkit/user.py]: Use constant variable rather than the integer

File: testsuites/syncgateway/functional/tests/test_conflicts.py [hunk 3]
```diff
@@ -342,3 +344,72 @@ def test_invalid_revs_limit_with_allow_conflicts(params_from_base_test_setup, sg
...
+            auth=session,
+            attachment_name="sample_text.txt"
+        )
+
+    update_from_sg_task.result()
+
+    # 4. Verify doc with latest attachment exists on SGW and CBS
+    sg_docs = sg_client.get_all_docs(url=sg_admin_url, db=sg_db, include_docs=True)["rows"]
+    for doc in sg_docs:
+        print "doc in sg_docs", doc
```
Comment (mgcouchbase) [inline testsuites/syncgateway/functional/tests/test_conflicts.py]: remove the print /replace with the info log

File: testsuites/syncgateway/functional/tests/test_users_channels.py [hunk 4]
```diff
@@ -277,3 +281,47 @@ def test_single_user_single_channel(params_from_base_test_setup, sg_conf_name):
...
+    sg_conf = sync_gateway_config_path_for_mode(sg_conf_name, mode)
+
+    cluster = Cluster(config=cluster_conf)
+    cluster.reset(sg_config_path=sg_conf)
+
+    sg_client = MobileRestClient()
+    if sync_gateway_version >= "2.5.0":
+        expvars = sg_client.get_expvars(sg_admin_url)
+        warn_count = expvars["syncgateway"]["global"]["resource_utilization"]["warn_count"]
+        print "warn_count is ", warn_count
```
Comment (mgcouchbase) [inline testsuites/syncgateway/functional/tests/test_users_channels.py]: remove print here

File: testsuites/syncgateway/upgrade/testserver/conftest.py [hunk 5]
```diff
@@ -323,7 +323,7 @@ def params_from_base_suite_setup(request):
         log_info("Running tests not using views")
         # Disable sg views in cluster configs
         persist_cluster_config_environment_prop(cluster_config, 'sg_use_views', False)
-        
+
```
Comment (mgcouchbase) [inline testsuites/syncgateway/upgrade/testserver/conftest.py]: no changes , revert this file

File: testsuites/syncgateway/functional/topology_specific_tests/multiple_servers/conftest.py [hunk 6]
```diff
@@ -62,6 +66,7 @@ def params_from_base_suite_setup(request):
     log_info("sa_installer_type: {}".format(sa_installer_type))
     log_info("delta_sync_enabled: {}".format(delta_sync_enabled))
     log_info("sg_platform: {}".format(sg_platform))
+    log_info("Delta_sync: {}".format(delta_sync_enabled))
```
Comment (mgcouchbase) [inline testsuites/syncgateway/functional/topology_specific_tests/multiple_servers/conftest.py]: Duplicate info

File: testsuites/syncgateway/functional/topology_specific_tests/multiple_servers/conftest.py [hunk 7]
```diff
@@ -41,10 +41,14 @@ def params_from_base_suite_setup(request):
     sa_installer_type = request.config.getoption("--sa-installer-type")
     delta_sync_enabled = request.config.getoption("--delta-sync")
     sg_platform = request.config.getoption("--sg-platform")
+    delta_sync_enabled = request.config.getoption("--delta-sync")
```
Comment (mgcouchbase) [inline testsuites/syncgateway/functional/topology_specific_tests/multiple_servers/conftest.py]: duplicate param

File: testsuites/syncgateway/functional/topology_specific_tests/load_balancer/conftest.py [hunk 8]
```diff
@@ -38,10 +38,14 @@ def params_from_base_suite_setup(request):
     sa_installer_type = request.config.getoption("--sa-installer-type")
     delta_sync_enabled = request.config.getoption("--delta-sync")
     sg_platform = request.config.getoption("--sg-platform")
+    delta_sync_enabled = request.config.getoption("--delta-sync")
```
Comment (mgcouchbase) [inline testsuites/syncgateway/functional/topology_specific_tests/load_balancer/conftest.py]: duplicate param  and info log

File: testsuites/syncgateway/functional/tests/test_sync.py [hunk 9]
```diff
@@ -503,3 +508,45 @@ def test_sync_require_roles(params_from_base_test_setup, sg_conf_name):
...
+    sg_client.create_user(url=sg_admin_url, db=sg_db, name='autotest', password='pass', channels=channels)
+    session = sg_client.create_session(url=sg_admin_url, db=sg_db, name='autotest', password='pass')
+
+    sg_doc_body = doc_generators.doc_size_byBytes(22000000)
+    doc_id = "20mb_doc"
+    doc = document.create_doc(doc_id, content=sg_doc_body, channels=channels)
+    try:
+        sg_client.add_doc(url=sg_url, db=sg_db, doc=doc, auth=session)
+    except HTTPError as h:
+        assert "413 Client Error: Request Entity Too Large for url" in h.message, "did not throw 413 client error"
```
Comment (mgcouchbase) [inline testsuites/syncgateway/functional/tests/test_sync.py]: h is not defined

**Comment by euniceatcouchbase** [comment]
there are 2 files contain conflicts, please resolve it before merge

File: libraries/provision/ansible/playbooks/restart-sync-gateway.yml [hunk 10]
```diff
@@ -1,6 +1,6 @@
 ---
 - hosts: sync_gateways
-  
+
```
[the remark this answers is not in the document]
Comment (sridevi-15) [reply libraries/provision/ansible/playbooks/restart-sync-gateway.yml]: it is intentional to add empty line after the hosts

File: testsuites/syncgateway/upgrade/testserver/conftest.py (hunk 5 above)
Comment (sridevi-15) [reply testsuites/syncgateway/upgrade/testserver/conftest.py]: intentional to have better format

File: testsuites/syncgateway/functional/tests/test_sync.py (hunk 9 above)
Comment (sridevi-15) [reply testsuites/syncgateway/functional/tests/test_sync.py]: h is defined for HTTPError. It is python way of defining

File: libraries/testkit/user.py [hunk 11]
```diff
@@ -186,9 +187,10 @@ def add_bulk_docs(self, doc_ids, retries=False):
         # Return list of cache docs that were added
         return bulk_docs_ids
 
-    def add_docs(self, num_docs, bulk=False, name_prefix=None, retries=False):
+    def add_docs(self, num_docs, bulk=True, name_prefix=None, retries=False):
 
         errors = list()
+        print "add docs for each user object"
```
[the remark this answers is not in the document]
Comment (sridevi-15) [reply libraries/testkit/user.py]: removed

File: testsuites/syncgateway/functional/tests/test_log_rotation_new.py (hunk 1 above)
Comment (sridevi-15) [reply testsuites/syncgateway/functional/tests/test_log_rotation_new.py]: done

File: libraries/testkit/user.py (hunk 2 above)
Comment (sridevi-15) [reply libraries/testkit/user.py]: done

File: testsuites/syncgateway/functional/tests/test_conflicts.py (hunk 3 above)
Comment (sridevi-15) [reply testsuites/syncgateway/functional/tests/test_conflicts.py]: done

File: testsuites/syncgateway/functional/tests/test_users_channels.py (hunk 4 above)
Comment (sridevi-15) [reply testsuites/syncgateway/functional/tests/test_users_channels.py]: done

File: testsuites/syncgateway/functional/topology_specific_tests/multiple_servers/conftest.py (hunk 6 above)
Comment (sridevi-15) [reply testsuites/syncgateway/functional/topology_specific_tests/multiple_servers/conftest.py]: some conftest files are missing delta sync. Looks like I added to conftest files which already existing . Removed duplicate lines

File: testsuites/syncgateway/functional/topology_specific_tests/multiple_servers/conftest.py (hunk 7 above)
Comment (sridevi-15) [reply testsuites/syncgateway/functional/topology_specific_tests/multiple_servers/conftest.py]: Removed duplicate lines

File: testsuites/syncgateway/functional/topology_specific_tests/load_balancer/conftest.py (hunk 8 above)
Comment (sridevi-15) [reply testsuites/syncgateway/functional/topology_specific_tests/load_balancer/conftest.py]: Removed duplicate lines

File: testsuites/syncgateway/functional/topology_specific_tests/load_balancer/conftest.py [hunk 12]
```diff
@@ -59,6 +63,7 @@ def params_from_base_suite_setup(request):
     log_info("sa_installer_type: {}".format(sa_installer_type))
     log_info("delta_sync_enabled: {}".format(delta_sync_enabled))
     log_info("sg_platform: {}".format(sg_platform))
+    log_info("Delta_sync: {}".format(delta_sync_enabled))
```
[the remark this answers is not in the document]
Comment (sridevi-15) [reply testsuites/syncgateway/functional/topology_specific_tests/load_balancer/conftest.py]: Removed duplicate lines

**Pull request merged**